lib/tpm2.c

```diff
@@ -2245,9 +2253,9 @@ tool_rc tpm2_hierarchycontrol(ESYS_CONTEXT *esys_context,
         goto tpm2_hierarchycontrol_skip_esapi_call;
     }
 
-
     TSS2_RC rval = Esys_HierarchyControl(esys_context, auth_hierarchy->tr_handle,
-            shandle, ESYS_TR_NONE, ESYS_TR_NONE, enable, state);
+            shandle, ESYS_TR_NONE, ESYS_TR_NONE,
+            fix_esys_hierarchy(enable), state);
```

Comment:
make these separate calls, and return a proper error. just passing a magic ~0 seems like a bad idea. The error occurs in one place with the LOGE but the failure will be in the actual ESYS calls ... seems odd

Reply:
See above, this is more for debugging purposes.
I did not want to change the function prototype from what it was before.
I guess it could be `TSS2_RC fix_esys_hierarchy(TPM2_HANDLE in, ESYS_TR *out);` and then we'd have actual explicit error handling, but that's going to be some more code...

Reply:
Wrong mistake here, it would be `TSS2_RC fix_esys_hierarchy(TPM2_HANDLE in, uint32_t *out);` of course...

Reply:
Im not tracking, we're not changing any function prototypes I was thinking:
```C

   TSS2_RC rval = fix_esys_hierarchy(enable, &enable)
   if (rval != TSS2_RC_SUCCESS) {
       LOG_ERR();
       return rc/goto;
   }

    rval = Esys_HierarchyControl(esys_context, auth_hierarchy->tr_handle,
            shandle, ESYS_TR_NONE, ESYS_TR_NONE, enable, state);
            shandle, ESYS_TR_NONE, ESYS_TR_NONE,
            enable, state);
```

Reply:
I know what you thought...

What I meant: I did not create `fix_esys_hierarchy()`. That function was already in there and I did not want to change its signature.

If you prefer that I can do it of course...